### PR TITLE
Improve the devcontainer settings

### PR DESCRIPTION
**Pull request by haifeng-jin**
This PR improves the devcontainer setup.
* Use the image specially prepared for devcontainers and install tensorflow on post commands.
* Extracted `isort` and `black` settings to `pyproject.toml` to make consistent behavior no matter how the commands are run. (Note that `pyproject.toml` is just there for the formatting setup not for package releases).
* Enable the GitHub CLI `gh` inside devcontainers.
* Fix the bug that `settings` in `devcontainer.json` did not work.
* Add a verticle ruler in the editor for line length limit.

### CODE REVIEW

File: shell/lint.sh [hunk 1]
```diff
@@ -31,7 +31,7 @@ then
   exit 1
 fi
 [ $# -eq 0 ] && echo "no issues with flake8"
-black --line-length 80 --check $files
```
Comment (ianstenbit) [inline shell/lint.sh:34]: Just to verify -- if using `lint.sh` outside of devcontainer, this setting will still be taken from `pyproject.toml` -- correct?

Comment (haifeng-jin) [reply shell/lint.sh:34]: Yes, that's correct.

**Comment by haifeng-jin** [comment]
> This all looks reasonable to me, but I have to admit that I don't use the Devcontainer or VSCode -- perhaps I should be using them?

You can try. I use chromebook so not many IDEs are available. I switched from vim and VS code has all the features I need.

**Review by ianstenbit — APPROVED** [review]
Thanks Haifeng!

**Pull request merged**